app_legacy/Helpers/database/user-activity.php

```diff
@@ -380,6 +380,11 @@ function getRecentlyPlayedGames(string $user, int $offset, int $count, ?array &$
             $recentlyPlayedGameIDs[] = $recentlyPlayedGame['GameID'];
         }
 
+        // cache may remember more than was asked for
+        if ($count < count($recentlyPlayedGameIDs)) {
+            $recentlyPlayedGameIDs = array_slice($recentlyPlayedGameIDs, 0, $count);
+        }
```

Comment:
if `API_GetUserSummary` was called asking for fewer than 5 games, it would still return 5 as we cache the 5 most recent games for the user page.

getting the twitch plugin to only ask for the most recent game (if it's not already) would eliminate fetching data for the other four most recent games.

Reply:
If all the plugin cares about is recent unlocks, perhaps we could get them to use a slimmer API? API_GetAchievementsEarnedBetween already exists, but requires the client to specify a time range.

I've proposed a new API in #1493 that's basically the same as API_GetAchievementsEarnedBetween but you specify the number of minutes to look back.